File: repos_estudiantes_ALVAREZ_tercera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA 3 INF-121/GENERECIDAD/EJERCICIO 1/Caja/src/Generecidad/Caja.java
package Generecidad;

public class Caja<T> {
    private T contenido;

    public void guardar(T valor) {
        this.contenido = valor;
    }

    public T obtener() {
        return contenido;
    }

    public static void main(String[] args) {
        Caja<Integer> cajaEntero = new Caja<>();
        cajaEntero.guardar(123);

        Caja<String> cajaTexto = new Caja<>();
        cajaTexto.guardar("Hola mundo");

        System.out.println("Caja de enteros: " + cajaEntero.obtener());
        System.out.println("Caja de texto: " + cajaTexto.obtener());
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/ALBA_PLATA_BRYAN_ALBERTO/PracticaTRES/GENERECIDAD/ejercicio1/caja.py
class Caja:
    def __init__(self):
        self.objeto = None

    def guardar(self, valor):
        self.objeto = valor

    def obtener(self):
        return self.objeto

caja1 = Caja()
caja1.guardar("comida")

caja2 = Caja()
caja2.guardar(123)

print("Contenido de caja1:", caja1.obtener())
print("Contenido de caja2:", caja2.obtener())